### PR TITLE
#163388261 feature user notifications

### PR DESCRIPTION
**Pull request by unah254**
#### What does this PR do?
Have user notifications
#### Description of Task to be completed?
- write unit tests
- add email notification handler
- add email template
- add serializers and views
- add verbs
- add comment, article and follow handlers
#### How should this be manually tested?
1. Checkout to the branch ```ft-user-receive-notifications-163388261```
2. `pip install -r requirements.txt`
3. `python manage.py runserver`
4. On postman signup two users:POST url-`http://127.0.0.1:8000/api/users/`

    ```
            {
               "username":"yourusername",
		"email": "youremail",
		"password":"yourpassword"		
            }
    ```

Login the registered users.POST url-`http://127.0.0.1:8000/api/users/login/`

As a user, follow the other registered user. POST  url-`http://127.0.0.1:8000/api/profiles/username/follow/`

           **N.B:** the other usershould receive an email notification that he has been followed

As the other user, create an article . POST url-`http://127.0.0.1:8000/api/articles/`
  
    **N.B** notification should be sent to you that an article has been created

Method | Endpoint | Functionality
--- | --- | ---
GET | `http://127.0.0.1:8000/api/notifications + 'token'` | View all notifications
PUT | `http://127.0.0.1:8000/api/notifications/unsubscribe/ + 'token'` | Unsubscribe from notifications
POST | `http://127.0.0.1:8000/api/notifications/subscribe/ + 'token'` | Subscribe to notifications
POST | `http://127.0.0.1:8000/api/profiles/username/follow/ + 'token'` | Get email notification for follow
POST | `http://127.0.0.1:8000/api/articles/ + 'token'` | Get email notification for article created
POST | `http://127.0.0.1:8000/api/articles/slug/comments/ + 'token'` | Get email notification for comment on your article

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388261](https://www.pivotaltracker.com/n/projects/2240063)
#### Screenshots (if appropriate)
followed notification
image

comment notification
image
get all notifications
image

### CODE REVIEW

File: authors/apps/usernotifications/views.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                user.app_notification_subscription = setting_data['app']
+
+            user.save()
+        message = {
+            "message": "You have successfully subscribed to notifications",
+            "data": serializer.data
+        }
+        return Response(message, status=status.HTTP_200_OK)
+
+    def put(self, request):
```
Comment (sam-karis) [inline authors/apps/usernotifications/views.py]: This is not necessary the post method above should handle both Subscribe and UnSubscribe.

File: authors/apps/usernotifications/urls.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        views.SubscribeUnsubscribeAPIView.as_view(),
+        name="email-unsubscribe"
+        ),
+    path(
+        'notifications/',
+        views.AllNotificationsAPIview.as_view(),
+        name="all-notifications"
+        ),
+    path(
+        'notifications/subscribe/',
```
Comment (sam-karis) [inline authors/apps/usernotifications/urls.py:13]: I think it would be better if you used one endpoint for both Subscribe and UnSubscribe.

**Review by sam-karis — CHANGES_REQUESTED** [review]
Good work @unah254 .

**Review by bryan-munene — COMMENTED** [review]
The Feature works as expected. However, fix the failing checks and test it on heroku.

**Review by sam-karis — APPROVED** [review]
Good work @unah254

**Comment by TooColline** [comment]
LGTM

**Pull request merged**